addons/account/models/account_move.py

```diff
@@ -126,17 +126,34 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
     def create(self, vals):
         move = super(AccountMove, self.with_context(check_move_validity=False, partner_id=vals.get('partner_id'))).create(vals)
         move.assert_balanced()
+
+        if 'line_ids' in vals:
+            move.update_lines_tax_exigibility()
+
         return move
 
     @api.multi
     def write(self, vals):
         if 'line_ids' in vals:
             res = super(AccountMove, self.with_context(check_move_validity=False)).write(vals)
             self.assert_balanced()
+            self.update_lines_tax_exigibility()
         else:
             res = super(AccountMove, self).write(vals)
+
         return res
 
+    @api.multi
+    def update_lines_tax_exigibility(self):
+        if all(account.user_type_id.type not in {'payable', 'receivable'} for account in self.mapped('line_ids.account_id')):
+            self.line_ids.write({'tax_exigible': True})
+        else:
+            tax_lines_caba = self.line_ids.filtered(lambda x: x.tax_line_id.tax_exigibility == 'on_payment')
+            base_lines_caba = self.line_ids.filtered(lambda x: any(tax.tax_exigibility == 'on_payment' for tax in x.tax_ids) )
```

Comment:
`x.tax_ids.flatten_taxes_hierarchy()`

Reply:
@smetl in v11? :thinking:

Reply:
I don't remember if you need to expand the group of taxes...

Reply:
I didn't file any pre-made stuff for that in v11, so I changed the code a little bit to handle groups of taxes ;)